File: 2406-decode-the-message/decode-the-message.py
class Solution(object):
    def decodeMessage(self, key, message):
        """
        :type key: str
        :type message: str
        :rtype: str
        """
        substitution_table = {}
        key_index = 0
        
        for char in key:
            if char != ' ' and char not in substitution_table:
                substitution_table[char] = chr(ord('a') + key_index)
                key_index += 1
        
        decoded_message = ''
        
        for char in message:
            if char == ' ':
                decoded_message += ' '
            elif char in substitution_table:
                decoded_message += substitution_table[char]
        
        return decoded_message

File: 2406-decode-the-message/decode-the-message.js
/**
 * @param {string} key
 * @param {string} message
 * @return {string}
 */
var decodeMessage = function(key, message) {
    let substitution_table = {};
    let key_index = 0;

    for (let char of key) {
        if (char !== " " && !(char in substitution_table)) {
            substitution_table[char] = String.fromCharCode('a'.charCodeAt(0) + key_index);
            key_index += 1;
        }
    }

    let decoded_message = "";

    for (let char of message) {
        if (char === " ") {
            decoded_message += " ";
        } else if (char in substitution_table) {
            decoded_message += substitution_table[char];
        }
    }

    return decoded_message;
};